ds3-autogen-python/src/main/java/com/spectralogic/ds3autogen/python/generators/request/PutObjectRequestGenerator.java

```diff
@@ -30,18 +35,35 @@
     private static final String PAYLOAD_NAME = "real_file_name";
 
     /**
-     * Gets the request payload model that handles string and stream payload types
+     * Gets the sorted list of optional constructor params, including the request payload
      */
     @Override
-    public RequestPayload toRequestPayload(final Ds3Request ds3Request, final String requestName) {
-        final String assignment = "self.object_name = typeCheckString(object_name)\n" +
+    public ImmutableList<ConstructorParam> toOptionalConstructorParams(final Ds3Request ds3Request) {
+        final ImmutableList.Builder<Arguments> builder = ImmutableList.builder();
+        builder.addAll(toOptionalArgumentsList(ds3Request.getOptionalQueryParams()));
+        builder.add(new Arguments("string", PAYLOAD_NAME));
+        builder.add(new Arguments("", "headers"));
+
+        return builder.build().stream()
+                .sorted(new CustomArgumentComparator())
+                .map(arg -> new ConstructorParam(camelToUnderscore(arg.getName()), true))
+                .collect(GuavaCollectors.immutableList());
+    }
+
+    /**
+     * Gets the python code that handles processing the request payload and headers
+     */
+    @Override
+    public String getAdditionalContent(final Ds3Request ds3Request, final String requestName) {
```

Comment:
We probably want to change how this and the getObject methods work in the Python SDK so that a user can pass in their own fd if they want to instead of us creating it for them.